.github/workflows/nix.yml

```diff
@@ -7,8 +7,10 @@ on:
       - nixpkgs-unstable
   pull_request:
   workflow_dispatch:
+concurrency: "{{ github.head_ref ||  github.ref }}"
 jobs:
   build:
+    timeout-minutes: 10080
```

Comment:
https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idtimeout-minutes